SU2_PY/fsi_computation.py

```diff
@@ -73,81 +70,81 @@ def main():
-	  print("Setting working directory : {}".format(os.getcwd()))
-      else: 
-	  print("Working directory is set to {}".format(os.getcwd()))
+        sys.path.append(os.getcwd())
+        print("Setting working directory : {}".format(os.getcwd()))
+      else:
+        print("Working directory is set to {}".format(os.getcwd()))
+
+  if have_MPI:
+    comm.barrier()
 
   # starts timer
   start = timer.time()
 
   confFile = str(options.filename)
 
-  FSI_config = FSI.io.FSIConfig(confFile) 		# FSI configuration file
+  FSI_config = FSI.FSIConfig(confFile) 		# FSI configuration file
   CFD_ConFile = FSI_config['CFD_CONFIG_FILE_NAME']	# CFD configuration file
   CSD_ConFile = FSI_config['CSD_CONFIG_FILE_NAME']	# CSD configuration file
 
   CSD_Solver = FSI_config['CSD_SOLVER']			# CSD solver
 
-  if have_MPI == True:
+  if have_MPI:
     comm.barrier()
 
   # --- Initialize the fluid solver --- #
   if myid == rootProcess:
     print('\n***************************** Initializing fluid solver *****************************')
   try:
-    FluidSolver = pysu2.CFluidDriver(CFD_ConFile, 1, FSI_config['NDIM'], comm)
+    FluidSolver = pysu2.CSinglezoneDriver(CFD_ConFile, 1, comm)
   except TypeError as exception:
-    print('A TypeError occured in pysu2.CSingleZoneDriver : ',exception)
-    if have_MPI == True:
+    print('A TypeError occured in pysu2.CSinglezoneDriver : ',exception)
+    if have_MPI:
       print('ERROR : You are trying to initialize MPI with a serial build of the wrapper. Please, remove the --parallel option that is incompatible with a serial build.')
     else:
       print('ERROR : You are trying to launch a computation without initializing MPI but the wrapper has been built in parallel. Please add the --parallel option in order to initialize MPI for the wrapper.')
     return
 
-  if have_MPI == True:
+  if have_MPI:
     comm.barrier()
-  
+
   # --- Initialize the solid solver --- # (!! for now we are using only serial solid solvers)
   if myid == rootProcess:
     print('\n***************************** Initializing solid solver *****************************')
-    if CSD_Solver == 'METAFOR':
-      from MetaforSolver import MtfSolver
-      SolidSolver = MtfSolver(CSD_ConFile)
-    elif CSD_Solver == 'NATIVE':
-      import NativeSolid
-      SolidSolver = NativeSolid.NativeSolidSolver(CSD_ConFile, True)
-    elif CSD_Solver == 'GETDP':
-      import GetDPSolver
-      SolidSolver = GetDPSolver.GetDPSolver(CSD_ConFile, True)
-    elif CSD_Solver == 'TESTER':
-      SolidSolver = FSI.PitchPlungeAirfoilStructuralTester.Solver(CSD_ConFile)
+    if CSD_Solver == 'AEROELASTIC':
+      from SU2_Nastran import pysu2_nastran
+      SolidSolver = pysu2_nastran.Solver(CSD_ConFile,False)
+    elif CSD_Solver == 'IMPOSED':
+      from SU2_Nastran import pysu2_nastran
+      SolidSolver = pysu2_nastran.Solver(CSD_ConFile,True)
+    else:
+      print("\n I have no idea about the solid solver you want to use")
```

Comment:
Did any of these solvers work? Maybe the error message could be a bit more robot-like "Invalid solid solver option" or something like that.

Reply:
I forgot about the message, I set it at the very beginning of my project and then it remained like that. 
The other structural solvers, except for the pitchplungeairfoil, were not present. I do not know them so I am not sure they would work. In any case, I had to do some modifications to the interface to update it so I am not sure they would still work.